crates/testenv/src/tx_template.rs

```diff
@@ -12,25 +12,27 @@ use bdk_chain::{
     Anchor,
 };
 use rand::distributions::{Alphanumeric, DistString};
+use std::borrow::Cow;
 use std::collections::HashMap;
 
 /// Template for creating a transaction in `TxGraph`.
 ///
 /// The incentive for transaction templates is to create a transaction history in a simple manner to
 /// avoid having to explicitly hash previous transactions to form previous outpoints of later
 /// transactions.
-#[derive(Clone, Copy, Default)]
-pub struct TxTemplate<'a, A> {
+#[derive(Clone, Default)]
+pub struct TxTemplate<A> {
     /// Uniquely identifies the transaction, before it can have a txid.
-    pub tx_name: &'a str,
-    pub inputs: &'a [TxInTemplate<'a>],
-    pub outputs: &'a [TxOutTemplate],
-    pub anchors: &'a [A],
+    pub tx_name: Cow<'static, str>,
+    pub inputs: Vec<TxInTemplate>,
+    pub outputs: Vec<TxOutTemplate>,
+    pub anchors: Vec<A>,
```

Comment:
@tvpeter Any reason/tradeoff on using `Vec`, instead of `Cow<'_, Tx..>`?

Reply:
Looking at it again, there is no trade-off. Thank you for the feedback. I will update it, as `Cow` seems more efficient.